evaluator.js

```diff
@@ -622,16 +620,26 @@ function is_empty_environment(env) {
    return is_null(env);
 }
 
+// set_value is used for setting a given value
+// at the head of given list, as well as to
+// store a boolean indicating whether the name of
+// the given value was declared as a constant or not
+
+function set_value(vals, val, is_constant) {
+    set_head(head(vals), val);
+    set_tail(vals, append(tail(vals), is_constant));
```

Comment:
Should this line be `set_tail(head(vals), is_constant)` ?

Reply:
The boolean indicating a constant declaration was being stored in another segment of `vals`. I've updated it as per your suggestion, it is now stored together with the value. This simplifies the code a little.